ldelf/ta_elf_rel.c

```diff
@@ -185,6 +202,12 @@ static void e32_relocate(struct ta_elf *elf, unsigned int rel_sidx)
 
 		str_tab_idx = shdr[sym_tab_idx].sh_link;
 		if (str_tab_idx) {
+			if (str_tab_idx >= elf->e_shnum)
+				err(TEE_ERROR_GENERIC,
```

Comment:
`TEE_ERROR_BAD_FORMAT` would probably be better, but I see `TEE_ERROR_GENERIC` is used in similar places, so if we want to change it it should be in a separate patch.

Reply:
Sounds good. I'll follow up this PR with another to make error codes more consistent.